soroban-sdk/src/deploy.rs

```diff
@@ -11,15 +11,18 @@
 //!
 //! ### Examples
 //!
+//! #### Deploy a contract without constructor (or 0-argument constructor)
+//!
 //! ```
 //! # use soroban_sdk::{contract, contractimpl, BytesN, Env, Symbol};
 //! #
+//! # const DEPLOYED_WASM: &[u8] = include_bytes!("../doctest_fixtures/contract.wasm");
 //! # #[contract]
 //! # pub struct Contract;
 //! #
 //! # #[contractimpl]
 //! # impl Contract {
-//! #     pub fn f(env: Env, wasm_hash: BytesN<32>) {
+//! #     pub fn deploy(env: Env, wasm_hash: BytesN<32>) {
 //! #         let salt = [0u8; 32];
 //! #         let deployer = env.deployer().with_current_contract(salt);
 //! #         // Deployed contract address is deterministic and can be accessed
```

Comment:
It looks like all the lines of these examples are commented, which means in the docs the example looks like this, with no lines displayed:
<img width="984" alt="Screenshot 2024-09-16 at 9 17 47 PM" src="https://github.com/user-attachments/assets/f93aca7d-1586-4e61-9ac1-5c92f6afb26a">
Ref: https://docs.rs/soroban-sdk/21.7.0/soroban_sdk/deploy/index.html

This is an existing problem. Either in this PR or in a follow we should uncomment lines 17 to 33, the contract and it's implementation.

Same with the other deploy example being added below.

Reply:
Updated both, I think it's worth having additional test coverage for the new features.